portfolio/src/main/java/com/google/sps/servlets/AuthServlet.java

```diff
@@ -0,0 +1,56 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.stream.Collectors;
+import java.util.List;
+import java.util.Arrays;
+
+@WebServlet("/auth")
+public class AuthServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html");
+    UserService userService = UserServiceFactory.getUserService();
+    response.getWriter().print(userService.isUserLoggedIn() + "");
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String[] requestData = request.getReader().lines().collect(Collectors.joining(" ")).split("\\s+");
+    String[] userData = getUserData(requestData);
+    String email = userData[0];
+    String password = userData[1];
+    
+  }
+
+  private String[] getUserData(String[] requestData) {
+    List<String> dataList = Arrays.asList(requestData);
+    int emailValueIndex = dataList.indexOf("name=\"email\"") + 1;
+    int passwordValueIndex = dataList.indexOf("name=\"password\"") + 1;
+    return new String[] {
```

Comment:
This isn't a pattern in Google Java code.  When you find yourself returning multiple values from a method, and the relationship between the values is that they're all part of the same thing, that's often a sign that you actually want to represent that data with a POJO (Plain Old Java Object -- just a set of properties with getters/setters).

Alternatively, break it up into multiple method calls via one parameterized method:

private static final String EMAIL_KEY = "email"

String email = myMethod(EMAIL_KEY)

myMethod(String key) {
&nbsp;&nbsp;int index = dataList.indexOf("name=\"" + key + "\"") + 1;
&nbsp;&nbsp;return dataList.get(index);
}

and likewise for an arbitrary number of key/value pairs.